models/register.php

```diff
@@ -2,13 +2,141 @@
+            $email_message = $input_check->Get_email_errorMessage();
+            $password_message = $input_check->Get_password_errorMessage();
+            if($username_message || $email_message || $password_message){
+                $this->page_data['username_errorMessage'] = $username_message;
+                $this->page_data['email_errorMessage'] = $email_message;
+                $this->page_data['password_errorMessage'] = $password_message;
+            }else{
+                if ($input_check->Insert_Accounts()){
+                    header("Location: /registered");  
+                }else{
+                    $errormessage = $input_check->Get_errorMessage();
+                    if($errormessage[1] = 1062){
+                        echo "<script>alert('既に登録されているアカウントです！');</script>";
+                    }else{
+                        echo "<script>alert('登録エラー');</script>";
+                    }
+                }
+            }
+        }
+    }
+
+}
+
+class Account {
+    private $account_name;              // ユーザ名
+    private $email;                     // メールアドレス
+    private $password;                  // パスワード
+    private $errorflg;                  // エラーフラグ
+    private $errorMessage;              // 汎用エラー
+    private $account_name_errorMessage; // ユーザ名エラーメッセージ
+    private $email_errorMessage;        // メールアドレスエラーメッセージ
+    private $password_errorMessage;     // パスワードエラーメッセージ
+
+    function __construct(){
+        // 画面から送信された内容を取得
+        $this->account_name = $_POST['account_name'];
+        if(!isset($this->account_name)){
+            $this->account_name = '';
+        }
+        $this->email = $_POST['email'];
+        if(!isset($this->email)){
+            $this->email = '';
+        }
+        $this->password = $_POST['password'];
+        if(!isset($this->password)){
+            $this->password = '';
+        }
+        $this->Input_check();
+    }
+
+    function Insert_Accounts(){
+        $rtnFlg = 0;
+        try{
+            //データベース接続
+            $DB = new DB_Connect();
+            $pdo = $DB->getPDO();
+            // SQL準備
+            $stmt = $pdo->prepare('INSERT INTO Accounts (account_name, email, password_hash)VALUES(:account_name, :email, :password_hash)');
+            // プレースホルダの値をセット
+            $password_hash = password_hash($this->password, PASSWORD_DEFAULT);
+            $stmt->bindParam(':account_name',$this->account_name, PDO::PARAM_STR);
+            $stmt->bindParam(':email', $this->email, PDO::PARAM_STR);
+            $stmt->bindParam(':password_hash', $password_hash, PDO::PARAM_STR);
+            // ＳＱＬ実行部分
+            $check = $stmt->execute();
+            if($check){
+                echo "<script>alert('アカウントの登録に成功しました！');</script>";
+                $rtnFlg = 1;
+            }else{
+                echo "<script>alert('アカウントの登録に失敗しました。');</script>";
+                $rtnFlg = 0;
+            }
+        }catch(PDOException $e){
+            $this->errorMessage = array(

```

Comment:
私の感想としては、$errorMessageの内容が分かりづらいのではないかと感じます。
特に加工しないPDOExceptionの例外情報が欲しいのであれば、
$this->errorMessage = $eでも良いかと思います。

もう少し可読性を上げるための手段として、このようなものはどうでしょう？
```suggestion
            $this->errorMessage = '登録エラー';
            if($e->errorInfo[1] == 1062){
                        $this->errorMessage = '既に登録されているアカウントです！';
            }
```

Reply:
未加工の情報はログ出力する形式にしました。表示メッセージは上記の記述にしておきます。

Reply:
ＯＫです。確認しました。